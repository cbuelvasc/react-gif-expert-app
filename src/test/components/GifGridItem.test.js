import { shallow } from 'enzyme';
import GifGridItem from '../../components/GifGridItem';

describe('Test GifGridItem component', () => {

    const title = 'Title';
    const url = 'https://127.0.0.1:3000'

    let wrapper = shallow(<GifGridItem title={title} url={url} />);

    beforeEach(() => {
        wrapper = shallow(<GifGridItem title={title} url={url} />);
    });

    test('Component GifGridItem should show correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Component GifGridItem should show the title', () => {
        const text = wrapper.find('p').text().trim();
        expect(text).toBe(title);
    });

    test('Component GifGridItem should show the img with src and url', () => {
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    });

    test('Component GifGridItem should show the div with class', () => {
        const div = wrapper.find('div');
        expect(div.prop('className').includes('animate__fadeIn')).toBe(true);
    });
});