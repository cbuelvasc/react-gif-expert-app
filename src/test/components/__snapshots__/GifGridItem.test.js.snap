// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GifGridItem component Component GifGridItem should show correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Title"
    src="https://127.0.0.1:3000"
  />
  <p>
    Title
  </p>
</div>
`;
